app/frontend/src/features/messages/messagelist/InfiniteMessageLoader.tsx

```diff
@@ -66,11 +75,51 @@ export default function InfiniteMessageLoader({
     messageEl?.scrollIntoView();
   }, [isFetchingNextPage]);
 
-  //scroll to the bottom on page load
+  // Scroll to bottom on load
   useLayoutEffect(() => {
     if (!scrollRef.current) return;
     scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
-  }, [scrollRef]);
+  }, []);
+
+  //**  Keep place or keep at bottom on window resize (i.e. keyboard popup on mobile (hopefully)) **//
+  // Save the current height outside of the eventlistener
+  const [currentClientHeight, setCurrentClientHeight] = useState(0);
+  useLayoutEffect(() => {
+    scrollRef.current && setCurrentClientHeight(scrollRef.current.clientHeight);
+  });
+
+  useLayoutEffect(() => {
+    const updateMessagePosition = () => {
+      if (!scrollRef.current) return;
+      const currentlyAtBottom =
+        scrollRef.current.scrollHeight <=
+        scrollRef.current.scrollTop + currentClientHeight;
+
+      if (currentlyAtBottom) {
+        // Send to bottom on window resize
+        scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
+      }
+      setCurrentClientHeight(scrollRef.current.clientHeight);
+    };
+
+    if (!scrollRef.current) return;
+    window.addEventListener("resize", updateMessagePosition);
+    return () => window.removeEventListener("resize", updateMessagePosition);
+  }, [currentClientHeight]);
+
+  //** Scroll to the bottom on new message  **//
+  // Saving messages count to know whether a new message has been created
+  const { messageId } = children.props.messages[0];
```

Comment:
This isn't really idiosyncratic, and it's a bit flakey. Not sure a good alternative. Maybe listening for changes in `scrollHeight`? You could also use [`React.Children.count()`](https://reactjs.org/docs/react-api.html#reactchildren) and do something if the number changes, but that's also flakey (we don't know if messages are the immediate children. Maybe someone else has some other ideas.

Reply:
Maybe just adding another prop - `latestMessageId`

Reply:
`children` aren't meant to be used this way at all and using it like that completely defeats its point, since as soon as this has more than one element, this will break. The fact you're trying to do it by using `any` is a big code smell.

@lucaslcode I see you've done something similar with `earliestMessageId` but in the opposite direction (scrolling up)? So yeah, maybe passing `latestMessageId` down as a prop is the way to go 🤷 , at least it'll be a lot more reliable than this as we'll be able to compute that from the React Query data from the parent component that'll be using this.

Reply:
> Maybe listening for changes in `scrollHeight`?

Using scrollHeight isn't an option because of the infinite loader, if you scroll up, the scrollHeight changes, which send you down to the bottom, same as React.children.count(), because there seems to be 20 children loaded in a "full" infinite loader element. 

Where can I get/make `latestMessageIid` to pass it down to?